### ISSUE TITLE
Assignment of meteorological values inconsistent with CMAQ methods

### ISSUE THREAD

**Issue by jlbeidler** [CEMPD/SMOKE#101]
In 2019 CMAQ transitioned from using the bi-linear interpolation method for assigning the meteorological values at point sources (BMATVEC in I/O API) to using the meteorological values from the grid cell that contains the point source. This change is described in CMAQ pull request 547: https://github.com/USEPA/CMAQ_Dev/pull/547
This impacts plume rise calculation consistency between CMAQ and SMOKE, particularly on more coarse grids.

To align SMOKE with CMAQ, laypoint will need to be updated so that meteorological variables are assigned to a point source using the grid cell in which the point resides.
An accompanying change is to update the WSPD calculated in preplm and fire_preplm to use linearly interpolated uw and vw values from the dot grid. An example is in the PTMET.F from CMAQ:

```
                  PTMET_DATA( N )%UWIND(L,S) = 0.5*( LOC_UWIND(MYC, MYR, L)
     &                                       +       LOC_UWIND(MYC+1, MYR, L) )
               
                  PTMET_DATA( N )%VWIND(L,S) = 0.5*( LOC_VWIND(MYC, MYR, L)
     &                                       +       LOC_VWIND(MYC, MYR+1, L) )
```
Where MYC = source column and MYR = source row

**Comment by jlbeidler** [comment]
Found another issue where the layer face pressure is passed incorrectly to the preplume function. This is fixed in PR 102.

**Comment by hnqtran** [comment]
Hi @jlbeidler, seems like I don't have access to https://github.com/USEPA/CMAQ_Dev/pull/547
Could you please give me access to it?

**Comment by jlbeidler** [comment]
Unfortunately I don't have the rights to grant access to that PR. Here is the summary:

> Goal: Address and fix a simulation crash found on 2016 CONUS domain in debug mode relating to the algorithm used to obtain meteorology information of a point source to calculate plume rise. The algorithm proposed will no longer bi-linearly spatially interpolate met data from four neighboring cells (neighboring cells based on point source location), but instead uses the met data from the cell in which the point source resides to calculate plume rise.
> 
> Files Changed:
> modified: src/emis/emis/PTBILIN.F
> modified: src/emis/emis/PTMET.F
> modified: src/emis/emis/PT3D_DEFN.F
> deleted: src/plrise/smoke/ungridb2.f
> deleted: src/util/util/bmatvec.F
> 
>

**Comment by hnqtran** [comment]
- Tested src/point/laypoint.f modified by @jlbeidler and confirmed that SMOKE was compiled successfully and laypoint worked.
- Conducted test runs with PTEGU in the 2021 emission platform using original and modified laypoint programs. Verified that there are slightly differences in emis_mole* 3d output files between the two versions at layers above the 1st layers, which are expected.
- Merged [PR 102](https://github.com/CEMPD/SMOKE/pull/102) to SMOKE master branch.

**Comment by cseppan** [comment]
Included in SMOKE v5.1
